src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -399,25 +420,14 @@ class ClapJuceWrapper : public clap::helpers::Plugin<clap::helpers::Misbehaviour
         }
 
         info->in_place_pair = info->id;
-        info->channel_count = (uint32_t)clob.size();
+        info->channel_count = (uint32_t)busDefaultLayout.size();
 
-        if (clob.size() == 1)
+        if (busDefaultLayout == juce::AudioChannelSet::mono())
             info->port_type = CLAP_PORT_MONO;
-        else
+        else if (busDefaultLayout == juce::AudioChannelSet::stereo())
             info->port_type = CLAP_PORT_STEREO;
-
-        auto requested = processor->getBusesLayout();
-        if (clob.size() == 1)
-            requested.getChannelSet(isInput, (int)index) = juce::AudioChannelSet::mono();
-        if (clob.size() == 2)
-            requested.getChannelSet(isInput, (int)index) = juce::AudioChannelSet::stereo();
-
-        processor->setBusesLayoutWithoutEnabling(requested);
-
-        DBG("audioPortsInfo " << (isInput ? "INPUT " : "OUTPUT ") << (int)index << " '"
```

Comment:
I will miss seeing this in my clion messages when i attach to bws engine. Chuckle. All looks good to me! I'll merge it and then I'll update surge to include both this and the if one.